File: docs/breez-sdk/snippets/python/src/user_settings.py
import logging
from breez_sdk_spark import (
    BreezSdk,
    StableBalanceActiveLabel,
    UpdateUserSettingsRequest,
)


async def get_user_settings(sdk: BreezSdk):
    # ANCHOR: get-user-settings
    try:
        user_settings = await sdk.get_user_settings()

        print(f"User settings: {user_settings}")
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: get-user-settings

async def update_user_settings(sdk: BreezSdk):
    # ANCHOR: update-user-settings
    try:
        spark_private_mode_enabled = True
        await sdk.update_user_settings(
            request=UpdateUserSettingsRequest(
                spark_private_mode_enabled=spark_private_mode_enabled
            )
        )
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: update-user-settings

async def activate_stable_balance(sdk: BreezSdk):
    # ANCHOR: activate-stable-balance
    try:
        await sdk.update_user_settings(
            request=UpdateUserSettingsRequest(
                spark_private_mode_enabled=None,
                stable_balance_active_label=StableBalanceActiveLabel.SET(label="USDB")
            )
        )
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: activate-stable-balance

async def deactivate_stable_balance(sdk: BreezSdk):
    # ANCHOR: deactivate-stable-balance
    try:
        await sdk.update_user_settings(
            request=UpdateUserSettingsRequest(
                spark_private_mode_enabled=None,
                stable_balance_active_label=StableBalanceActiveLabel.UNSET()
            )
        )
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: deactivate-stable-balance

File: docs/breez-sdk/snippets/rust/src/user_settings.rs
use anyhow::Result;
use breez_sdk_spark::*;
use log::info;

pub(crate) async fn get_user_settings(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: get-user-settings
    let user_settings = sdk.get_user_settings().await?;
    info!("User settings: {:?}", user_settings);
    // ANCHOR_END: get-user-settings
    Ok(())
}

pub(crate) async fn update_user_settings(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: update-user-settings
    // Fields left as None are not changed. Settings that take an enum value
    // accept Set to assign one, or Unset to clear it.
    sdk.update_user_settings(UpdateUserSettingsRequest {
        spark_private_mode_enabled: Some(true),
        stable_balance_active_label: None,
        spark_master_identity_public_key: Some(SparkMasterIdentityPublicKey::Set {
            public_key: "<hex encoded public key>".to_string(),
        }),
    })
    .await?;
    // ANCHOR_END: update-user-settings
    Ok(())
}

pub(crate) async fn activate_stable_balance(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: activate-stable-balance
    sdk.update_user_settings(UpdateUserSettingsRequest {
        spark_private_mode_enabled: None,
        stable_balance_active_label: Some(StableBalanceActiveLabel::Set {
            label: "USDB".to_string(),
        }),
        spark_master_identity_public_key: None,
    })
    .await?;
    // ANCHOR_END: activate-stable-balance
    Ok(())
}

pub(crate) async fn deactivate_stable_balance(sdk: &BreezSdk) -> Result<()> {
    // ANCHOR: deactivate-stable-balance
    sdk.update_user_settings(UpdateUserSettingsRequest {
        spark_private_mode_enabled: None,
        stable_balance_active_label: Some(StableBalanceActiveLabel::Unset),
        spark_master_identity_public_key: None,
    })
    .await?;
    // ANCHOR_END: deactivate-stable-balance
    Ok(())
}
